packages/component/src/Icon/ComponentIcon.tsx

```diff
@@ -20,11 +24,23 @@ const componentIconPropsSchema = pipe(
 type ComponentIconProps = InferInput<typeof componentIconPropsSchema>;
 
 function BaseComponentIcon(props: ComponentIconProps) {
-  const { className } = validateProps(componentIconPropsSchema, props);
+  const {
+    className,
+    'aria-hidden': ariaHidden,
+    'aria-label': ariaLabel,
+    role
+  } = validateProps(componentIconPropsSchema, props);
 
   const classNames = useStyles(styles);
 
-  return <div className={cx(classNames['component-icon'], className)} />;
+  return (
+    <div
+      aria-hidden={ariaHidden}
+      aria-label={ariaLabel}
+      className={cx(classNames['component-icon'], className)}
+      role={role}
+    />
+  );
 }
 
 const ComponentIcon = createIconComponent(styles, BaseComponentIcon);
```

Comment:
Maybe need to add some typing to the return value of `createIconComponent` so it can infer `size` (and stuff) from CSS modules.